docs/src/components/ReanimatedAnimations/ReanimatedAnimationsSection/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState, useEffect } from 'react';
+import CodeBlock from '@theme/CodeBlock';
+import styles from './styles.module.css';
+import InteractiveExampleComponent from '../../InteractiveExample/InteractiveExampleComponent';
+import ArrowRight from '@site/static/img/examples/arrow-right-light.svg';
+
+interface Props {
+  title: string;
+  body: string;
+  docsLink: string;
+  component: React.ReactNode;
+  mobileComponent: React.ReactNode;
+  tabletComponent: React.ReactNode;
+  code: string;
+  idx: number;
+}
+
+export default function ReanimatedAnimationsSection({
+  title,
+  body,
+  docsLink,
+  component,
+  mobileComponent,
+  tabletComponent,
+  code,
+  idx,
+}: Props) {
+  const [isMobileView, setIsMobileView] = useState(window.innerWidth <= 420);
+  const [isTabletView, setIsTabletView] = useState(window.innerWidth <= 860);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobileView(window.innerWidth <= 420);
+      setIsTabletView(window.innerWidth <= 860 && window.innerWidth > 420);
+    };
+
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
+
+  return (
+    <div className={styles.animationsSectionWrapper}>
+      {mobileComponent !== null && !!isMobileView && !isTabletView && (
+        <div>
+          <InteractiveExampleComponent idx={idx} component={mobileComponent} />
+        </div>
+      )}
+      {tabletComponent !== null && !!isTabletView && (
+        <div>
+          <InteractiveExampleComponent idx={idx} component={tabletComponent} />
+        </div>
+      )}
+      {((!isMobileView && !isTabletView) ||
+        (!mobileComponent && !tabletComponent)) && (
+        <div>
+          <InteractiveExampleComponent idx={idx} component={component} />
+        </div>
+      )}
```

Comment:
I'd simplify that logic by extracting it to the component body some thing like (pseudocodeish)

```jsx
let componentToRender = component;

if (mobile condition) {
  componentToRender = mobileComponent;
}

if (tablet condition) {
 componentToRender = tabletComponent
}


return (
  <div>
    <InteractiveExampleComponent idx={idx} component={componentToRender} />
  </div>
   // rest of the markup
 )